BinaryObjectScanner/FileType/RAR.cs

```diff
@@ -91,5 +103,71 @@ public bool Extract(Stream? stream, string file, string outDir, bool lookForHead
             return false;
 #endif
         }
+#if NET462_OR_GREATER || NETCOREAPP
+        /// <inheritdoc cref="IExtractable.Extract(Stream?, string, string, bool)"/>
+        // Extraction method for non-solid archives. This iterates over each entry in the archive to extract every file
+        // individually, in order to extract all valid files from the archive.
+        public bool Extract_Non_Solid(ReaderOptions? readerOptions, RarArchive? rarFile, string file, string?
```

Comment:
Helper methods like this don't have to be `public`. `private` is fine unless they needed to be referenced elsewhere in the library.